### ISSUE TITLE
How do I opt out of deep-data merging in data files?

### ISSUE THREAD

**Issue by AleksandrHovhannisyan** [11ty/eleventy#2868]
## Description

The [11ty docs on opting out of data merging](https://www.11ty.dev/docs/data-deep-merge/#using-the-override-prefix) mention that you can use an `override:` prefix to opt out of merging data for a specific front-matter variable:

> Use the override: prefix on any data key to opt-out of this merge behavior for specific values or nested values.

```json
{
  "tags": ["posts"]
}
```

```md
---
override:tags: []
---
```

> Even though normally the posts/firstpost.md file would inherit the posts tag from the directory data file (per normal [data cascade rules](https://www.11ty.dev/docs/data/)), we can override the tags value to be an empty array to opt-out of this behavior.

But how would I opt out of deep data merging for data files? For example, suppose I have this file: `src/_data/data.js`:

```js
// src/_data/data.js
module.exports = {
  key: "value",
}
```

And later, somewhere in a directory, I want to override this data file:

```js
// src/_collection/collection.11tydata.js
module.exports = {
  "override:data": {
    key: "new-value"
  }
}
```

Unfortunately, this doesn't work. Is this not supported, or do I need to use a different syntax?

## Steps to repro

1. Clone https://github.com/AleksandrHovhannisyan/11ty-sandbox/tree/merge-override
2. Run `yarn` to install dependencies.
3. Run `yarn serve`.
4. Observe that `{ "key1": "value1", "key2": "value2" }` is logged in the server console.
5. Expectation: Only `{ "key2": "value2" }` gets logged.

**Comment by zachleat** [comment]
and just to be clear you don’t want to use `eleventyConfig.setDataDeepMerge(false);`?

**Comment by AleksandrHovhannisyan** [comment]
Correct, just opt out in this one instance but deep merge everywhere else.

**Comment by AleksandrHovhannisyan** [comment]
And sorry, the example I provided wasn't the best. Here's a better one:

**Current behavior**

```js
// src/_data/data.js
module.exports = {
  key1: "value1",
};
```

```js
// src/_collection/_collection.11tydata.js
module.exports = {
  "override:data": {
    key2: "value2",
  },
};
```

Result for `data`:

```json
{
  "key1": "value1",
  "key2": "value2"
}
```

Expected shape for data:

```json
{
  "key2": "value2"
}
```

**Comment by AleksandrHovhannisyan** [comment]
Hey, just a quick follow up on this issue. I would be happy to put in a fix myself since I know you're busy. I've created a repro sandbox here that can be cloned: https://github.com/AleksandrHovhannisyan/11ty-sandbox/tree/merge-override. I've updated the original post here to mention repro steps.

I was looking through the code, and what I found confusing is that the tests seem to already account for this case:

https://github.com/11ty/eleventy/blob/ecd0579a2dded6939510b7c23841388b26eb6d16/test/MergeTest.js#L47

The only usage of `Merge` that I could find is here:

https://github.com/11ty/eleventy/blob/ecd0579a2dded6939510b7c23841388b26eb6d16/src/Plugins/RenderPlugin.js#L380-L384

In my example above, Merge would get called with these arguments once `data` is spread:

```js
Merge({}, { "key1": "value1", { "override:data": { key2: "value2" } });
```

So I wonder why that doesn't work...

**Comment by zachleat** [comment]
Marginally related #3937
